src/app/server/Server.cpp

```diff
@@ -593,6 +597,9 @@ void Server::ScheduleFactoryReset()
         GetInstance().GetFabricTable().DeleteAllFabrics();
         PlatformMgr().HandleServerShuttingDown();
         ConfigurationMgr().InitiateFactoryReset();
+#if CHIP_CONFIG_TC_REQUIRED
+        GetInstance().GetTermsAndConditionsProvider()->ResetAcceptance();
+#endif
```

Comment:
This should not be required due to the fabric delegate that listens for last fabric deletion, which will cause `GetInstance().GetFabricTable().DeleteAllFabrics();` above to actually have done the ResetAcceptance.